src/Judge/NTS.Judge/Core/Behinds/Adapters/RanklistBehind.cs

```diff
@@ -100,9 +102,17 @@ async Task SafeSelectRanking(int id)
     {
         var ranking = await _rankings.Read(id);
         GuardHelper.ThrowIfDefault(ranking);
+        CreateDocument(ranking);
+        EmitChange();
+    }
 
+    async void CreateDocument(Ranking ranking)
+    {
         Ranklist = new Ranklist(ranking);
-        EmitChange();
+        var enduranceEvent = await _events.Read(0);
+        GuardHelper.ThrowIfDefault(enduranceEvent);
+        var officials = await _officials.ReadAll();
+        Document = new RanklistDocument(Ranklist!, enduranceEvent, officials);
     }
```

Comment:
This is better, but that's not "exposing" a method. Exposing means you open something up to consumers i -i.e. make it public. My idea was that you can have this public method in the Protocol component where you will Create a document and print it. Instead of exposing a document property you will expose a method to create one